pets/pet_rock.py

```diff
@@ -36,16 +39,24 @@ def set_location(self, x,y):
     def get_location(self):
         return self.pet_location
 
-    def set_current_animation(self, animation_selected):
+    def set_play_once(self, is_playing_once):
+        self.is_play_once = is_playing_once
+
+    def set_current_animation(self, animation_selected, is_playing_once = False):
```

Comment:
We can now use this `is_playing_once` argument to play pet animations `once` or not